File: projects/gnomad/src/client/datasets.js
// Define Default dataset
export const getDefaultDataset = () => 'gnomad_r3';

// List of available datasets. 
const datasets = {
  gnomad_r3: {
    label: 'gnomAD v3',
    referenceGenome: 'GRCh38',
    isTopLevel: true },
  gnomad_r2_1: {
    label: 'gnomAD v2.1.1',
    referenceGenome: 'GRCh37',
    isTopLevel: true },
  gnomad_r2_1_controls: {
    label: 'gnomAD v2.1.1 (controls)',
    referenceGenome: 'GRCh37',
    isTopLevel: false },
  gnomad_r2_1_non_cancer: {
    label: 'gnomAD v2.1.1 (non-cancer)',
    referenceGenome: 'GRCh37',
    isTopLevel: false },
  gnomad_r2_1_non_neuro: {
    label: 'gnomAD v2.1.1 (non-neuro)',
    referenceGenome: 'GRCh37',
    isTopLevel: false },
  gnomad_r2_1_non_topmed: {
    label: 'gnomAD v2.1.1 (non-TOPMed)',
    referenceGenome: 'GRCh37',
    isTopLevel: false },
  gnomad_sv_r2_1: {
    label: 'gnomAD SVs v2.1',
    referenceGenome: 'GRCh37',
    isTopLevel: false },
  gnomad_sv_r2_1_controls: {
    label: 'gnomAD SVs v2.1 (controls)',
    referenceGenome: 'GRCh37',
    isTopLevel: false },
  gnomad_sv_r2_1_non_neuro: {
    label: 'gnomAD SVs v2.1 (non-neuro)',
    referenceGenome: 'GRCh37',
    isTopLevel: false },
  exac: {
    label: 'ExAC v1.0',
    referenceGenome: 'GRCh37',
    isTopLevel: true },
}

export const referenceGenomeForDataset = datasetId => datasets[datasetId].referenceGenome || 'GRCh38';

export const labelForDataset = datasetId => datasets[datasetId].label || 'Unknown'

export const topLevelDataset = () => Object.keys(datasets).filter(o => datasets[o].isTopLevel);

File: gnomad-go-api/internal/data/datasets.go
package data

type DatasetConfig struct {
	ID              string
	ReferenceGenome string
	HasExome        bool
	HasGenome       bool
	HasJoint        bool
}

var Datasets = map[string]DatasetConfig{
	"gnomad_r4": {
		ID:              "gnomad_r4",
		ReferenceGenome: "GRCh38",
		HasExome:        true,
		HasGenome:       true,
		HasJoint:        true,
	},
	"gnomad_r4_non_ukb": {
		ID:              "gnomad_r4_non_ukb",
		ReferenceGenome: "GRCh38",
		HasExome:        true,
		HasGenome:       true,
		HasJoint:        true,
	},
	"gnomad_r3": {
		ID:              "gnomad_r3",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r3_controls_and_biobanks": {
		ID:              "gnomad_r3_controls_and_biobanks",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r3_non_cancer": {
		ID:              "gnomad_r3_non_cancer",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r3_non_neuro": {
		ID:              "gnomad_r3_non_neuro",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r3_non_topmed": {
		ID:              "gnomad_r3_non_topmed",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r3_non_v2": {
		ID:              "gnomad_r3_non_v2",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r2_1": {
		ID:              "gnomad_r2_1",
		ReferenceGenome: "GRCh37",
		HasExome:        true,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r2_1_controls": {
		ID:              "gnomad_r2_1_controls",
		ReferenceGenome: "GRCh37",
		HasExome:        true,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r2_1_non_cancer": {
		ID:              "gnomad_r2_1_non_cancer",
		ReferenceGenome: "GRCh37",
		HasExome:        true,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r2_1_non_neuro": {
		ID:              "gnomad_r2_1_non_neuro",
		ReferenceGenome: "GRCh37",
		HasExome:        true,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r2_1_non_topmed": {
		ID:              "gnomad_r2_1_non_topmed",
		ReferenceGenome: "GRCh37",
		HasExome:        true,
		HasGenome:       true,
		HasJoint:        false,
	},
	"exac": {
		ID:              "exac",
		ReferenceGenome: "GRCh37",
		HasExome:        true,
		HasGenome:       false,
		HasJoint:        false,
	},
	"gnomad_sv_r2_1": {
		ID:              "gnomad_sv_r2_1",
		ReferenceGenome: "GRCh37",
		HasExome:        false,
		HasGenome:       false,
		HasJoint:        false,
	},
	"gnomad_sv_r2_1_controls": {
		ID:              "gnomad_sv_r2_1_controls",
		ReferenceGenome: "GRCh37",
		HasExome:        false,
		HasGenome:       false,
		HasJoint:        false,
	},
	"gnomad_sv_r2_1_non_neuro": {
		ID:              "gnomad_sv_r2_1_non_neuro",
		ReferenceGenome: "GRCh37",
		HasExome:        false,
		HasGenome:       false,
		HasJoint:        false,
	},
	"gnomad_sv_r4": {
		ID:              "gnomad_sv_r4",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       false,
		HasJoint:        false,
	},
	"gnomad_cnv_r4": {
		ID:              "gnomad_cnv_r4",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       false,
		HasJoint:        false,
	},
}
